### PR TITLE
Provide alternatives to ext-crypto

### PR DESCRIPTION
**Pull request by johanderuijter**
This commit makes ext-crypto an optional dependency.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,13 +12,21 @@
...
     "require-dev": {
-        "phpspec/phpspec": "~3.1"
+        "ext-crypto": "*",
+        "lib-openssl": "*",
+        "spomky-labs/php-aes-gcm": "~1.0",
+        "phpspec/phpspec": "~3.2",
+        "phpspec/prophecy": "~1.6"
+    },
+    "suggest": {
+        "ext-crypto": "Use exp-crypto to encrypt the notifications",
```
Comment (peternijssen) [inline composer.json]: Typo

File: src/Crypto/ExtCryptoCrypt.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+class ExtCryptoCrypt implements Crypt
+{
+    /**
+     * Constructor.
+     *
+     * @throws UnsupportedEncryptionMethod When ext-crypto is not installed of the version is incompatible with the version of php.
+     */
+    public function __construct()
+    {
+        if (!phpversion('crypto')) {
```
Comment (peternijssen) [inline src/Crypto/ExtCryptoCrypt.php:17]: Never knew you could check extensions with phpversion. always used extension_loaded()

**Comment by johanderuijter** [comment]
Fixed the typo. Did you find anything else?

**Comment by peternijssen** [comment]
Rest looked fine to me. Should we tag a new version, if so, what would it be?

**Pull request merged**